geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/sft/package.scala

```diff
@@ -26,41 +26,44 @@ package object sft {
   //  user data keys and values may be mutable objects, so while the user data map will not change,
   //  the values inside may
 
-  class ImmutableAttributeDescriptor(`type`: AttributeType,
-                                     name: Name,
-                                     minOccurs: Int,
-                                     maxOccurs: Int,
-                                     isNillable: Boolean,
-                                     defaultValue: AnyRef,
-                                     userData: java.util.Map[AnyRef, AnyRef])
-      extends AttributeDescriptorImpl(`type`, name, minOccurs, maxOccurs, isNillable, defaultValue) {
+  class ImmutableAttributeDescriptor(
+      `type`: AttributeType,
+      name: Name,
+      minOccurs: Int,
+      maxOccurs: Int,
+      isNillable: Boolean,
+      defaultValue: AnyRef,
+      ud: java.util.Map[_ <: AnyRef, _ <: AnyRef]
```

Comment:
Meh, any reason to abbreviate userData here?

Reply:
there's a protected/package level variable called 'userData' in one of the super-classes that makes intellij complain when we reference this one...

Reply:
Although maven seems to handle it ok either way